nginx/datadog_checks/nginx/nginx.py

```diff
@@ -83,7 +83,8 @@ def check(self, instance):
         if not use_plus_api:
             response, content_type, version = self._get_data(instance, url)
             # for unpaid versions
-            self._set_version_metadata(version)
+            if self.is_metadata_collection_enabled():
+                self._set_version_metadata(version)
```

Comment:
If we use the `@AgentCheck.metadata_entrypoint` decorator for `_set_version_metadata()` definition, we won't need to check the collection before attempting.